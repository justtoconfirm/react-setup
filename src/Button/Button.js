import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const Button = (props) => {

    const {
        type = 'button',
        url = '',
        className = '',
        theme = '',
        isFullWidth = false,
        children, 
        onClick
    } = props

    const buttonClass = classNames('btn', className, {
        'btn--primary' : theme === 'primary',
        'btn--secondary' : theme === 'secondary',
        'btn--full-width' : isFullWidth
    })

    const isAnchor = url && (url.includes('http') || url.startsWith('#') || url.startsWith('mailto') || url.startsWith('/'))

    const renderAnchor = () =>
        <a href={url} className={buttonClass}>{children}</a>

    const renderButton = () => 
        <button {...{type, onClick}} className={buttonClass}>{children}</button>

    return (
        isAnchor ? renderAnchor() : renderButton()
    )
}

Button.propTypes = {
    type: PropTypes.string,
    url: PropTypes.string,
    className: PropTypes.string,
    theme: PropTypes.string,
    isFullWidth: PropTypes.bool,
    children: PropTypes.string,
    onClick: PropTypes.string
}

export default Button
