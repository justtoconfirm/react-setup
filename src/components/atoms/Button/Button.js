import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const Button = props => {
  const {
    type = 'button',
    url = '',
    className = '',
    theme = '',
    isFullWidth = false,
    children,
    onClick
  } = props

  const buttonClass = classNames('a-btn', className, {
    'a-btn--primary': theme === 'primary',
    'a-btn--secondary': theme === 'secondary',
    'a-btn--full-width': isFullWidth
  })

  const isAnchor = url && (url.includes('http') || url.startsWith('#') || url.startsWith('mailto') || url.startsWith('/'))

  const renderAnchor = () =>
    <a href={url} className={buttonClass}>{children}</a>

  const renderButton = () =>
    <button {...{ type, onClick }} className={buttonClass}>{children}</button>

  return (
    isAnchor ? renderAnchor() : renderButton()
  )
}

Button.propTypes = {
  type: PropTypes.string,
  url: PropTypes.string,
  className: PropTypes.string,
  theme: PropTypes.string,
  isFullWidth: PropTypes.bool,
  children: PropTypes.string,
  onClick: PropTypes.string
}

export default Button
