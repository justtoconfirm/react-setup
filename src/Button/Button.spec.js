import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('<Button/>', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<Button />)
    expect(wrapper.find('.a-btn').exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
