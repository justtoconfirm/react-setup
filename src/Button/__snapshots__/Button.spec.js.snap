// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render without throwing an error 1`] = `ShallowWrapper {}`;
